components/script/dom/bindings/str.rs

```diff
@@ -671,3 +673,18 @@ fn max_week_in_year(year: u32) -> u32 {
 fn is_leap_year(year: u32) -> bool {
     year % 400 == 0 || (year % 4 == 0 && year % 100 != 0)
 }
+
+/// https://html.spec.whatwg.org/multipage/#rules-for-parsing-floating-point-number-values
+fn parse_floating_point_number(input: &str) -> Result<String, ()> {
+    match input.parse::<f64>() {
+        Ok(val) if !(
+            // A valid number is the same as what rust considers to be valid,
+            // except for +1., NaN, and Infinity.
```

Comment:
This does not account for trimming ASCII whitespace, I am also not sure this properly accounts for steps 15-17 of the [spec](https://html.spec.whatwg.org/multipage/#rules-for-parsing-floating-point-number-values).